source/common/router/config_impl.cc

```diff
@@ -65,6 +65,16 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHost& vhost, const Json::Obj
   if (!(isRedirect() ^ !cluster_name_.empty())) {
     throw EnvoyException("routes must be either redirects or cluster targets");
   }
+
+  if (route.hasObject("headers")) {
+    std::vector<Json::Object> config_headers = route.getObjectArray("headers");
+    for (const Json::Object& header_map : config_headers) {
+      // allow header value to be empty, allows matching to be only based on header presence.
+      HeaderData header_data(header_map.getString("header_name"),
+                             header_map.getString("header_value", ""));
+      config_headers_.push_back(header_data);
```

Comment:
looks like emplace_back would not allow me to do [uniform initialization](http://stackoverflow.com/questions/10890653/why-would-i-ever-use-push-back-instead-of-emplace-back)